WikiWatershed/mmw-geoprocessing
# Add MapShed and Subbasin Tests

## Overview

We repurpose the benchmark scripts to save their outputs, and then add an optional `--verify` parameter that will compare the live results against these saved outputs.

Also add a test script that runs the `benchmark-mapshed` and `benchmark-subbasin` scripts with the `--verify` parmeter to run the tests.

Also add a ruff script to format the Python scripts, and runs it to modernize them.

I did not add any tests to `benchmark-sync` because between MapShed and Subbasin we're covering all relevant use cases.

Closes #108 

## Testing Instructions

- Check out this branch
- Run `./scripts/setup` to ensure you have the right Python setup
- Run `./scripts/server` to ensure you have the service running
- Run `./scripts/test` and ensure there are no errors
